pi_top_usb_setup/utils.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import shutil
+import tarfile
+from signal import SIGTERM, raise_signal
+from typing import Optional
+
+from pitop.common.command_runner import run_command
+
+logger = logging.getLogger(__name__)
+
+
+def systemctl(command: str, name: str, timeout=10) -> Optional[str]:
+    try:
+        cmd = f"systemctl {command} {name}.service"
+        logger.info(f"Executing '{cmd}'")
+        output = run_command(
+            cmd,
+            timeout=timeout,
+            check=False,
+        )
+        return output.strip(" \n")
+    except Exception as e:
+        logger.error(f"Error on systemctl(command={command}, name={name}): {e}")
+        return None
+
+
+def close_app() -> None:
+    if systemctl("is-active", "pt-usb-setup") == "active":
+        systemctl("stop", "pt-usb-setup")
+    else:
+        raise_signal(SIGTERM)
+
+
+def get_extracted_size(file: str) -> int:
```

Comment:
comment would be good. maybe with reference to stackoverflow or tar docs or wherever you learn this